keps/sig-node/2413-seccomp-by-default/README.md

```diff
@@ -0,0 +1,381 @@
+feature.
+
+For alternative proposals please refer to the [Alternatives
+section](#alternatives).
+
+### User Stories
+
+As a Kubernetes admin, I want to ensure that my cluster is secure by default
+without relying on workloads opting to use seccomp.
+
+### Risks and Mitigations
+
+Some workloads that may be running without seccomp may break with seccomp
+enabled by default. All workloads would have to be tested in a staging/test
+environment to ensure there are no breakages. Seccomp could either be turned off
+or custom profiles could be created for such workloads.
+
+The configuration possibilities of container runtimes differ in conjunction with
+seccomp. For example:
+
+- **containerd**
+  - can only use the internal default profile for `RuntimeDefault`
+  - can use a different profile for empty (unconfined) workloads via the
+    `unset_seccomp_profile` option
+- **CRI-O**
+  - can specify a different `RuntimeDefault` profile via the `seccomp_profile`
+    option
+  - can use `RuntimeDefault` for empty (unconfined) workloads via
+    `seccomp_use_default_when_empty`
+
+This can result in a behavioral change when doing cluster upgrades while runtime
+administrates may have to take action if they enable the feature.
+
+## Design Details
+
+The feature gate `SeccompDefault` will ensure that the API graduation can be
+done in the standard Kubernetes way. The implementation will be mainly a switch
+from `Unconfined` to `RuntimeDefault`.
+
+Documentation around the feature will be added to the [k/website seccomp
+section](https://kubernetes.io/docs/tutorials/clusters/seccomp).
+
+### Test Plan
+
+There will be unit tests for the feature, whereas the existing seccomp tests can
+be extended to cover the new behavior if enabled.
+
+### Graduation Criteria
+
+#### Alpha
+
+- [ ] Implement the new feature gate
+- [ ] Ensure proper tests are in place
+- [ ] Update documentation to make the feature visible
+
+#### Alpha to Beta Graduation
+
+- [ ] Enable the feature per default
+- [ ] No major bugs reported in the previous cycle
+
+#### Beta to GA Graduation
+
+- [ ] Allowing time for feedback (3 releases)
+- [ ] Risks have been addressed by every common container runtime
+
+### Upgrade / Downgrade Strategy
+
+It's recommended to test the existing workloads with the `RuntimeDefault`
+profile before turning the feature on. Beside that, the feature can be enabled
+on a per node basis to reduce the risk of failing production workloads.
+
+### Version Skew Strategy
+
+There is no explicit version skew strategy required because the feature acts as
+a toggle switch.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+- **How can this feature be enabled / disabled in a live cluster?**
+
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: `SeccompDefault`
+    - Components depending on the feature gate: `none`
```

Comment:
There has to be a component relying on this feature gate. Which binary receives this value and changes the default?

Reply:
kubelet

Reply:
Fixed as suggested :+1: